src/Mod/Part/App/AttachExtension.cpp

```diff
@@ -477,6 +482,164 @@ bool AttachExtension::extensionHandleChangedPropertyName(Base::XMLReader& reader
+        const App::Expression* exprY = nullptr;
+        const App::Expression* exprYaw = nullptr;
+        for (const auto& [oid, expr] : owner->ExpressionEngine.getExpressions()) {
+            if (oid == oidX) {
+                exprX = expr;
+            } else if (oid == oidY) {
+                exprY = expr;
+            } else if (oid == oidYaw) {
+                exprYaw = expr;
+            }
+        }
+
+        // convert placement and expressions according to the dominant axis
+        App::Expression* newExprX = nullptr;
+        App::Expression* newExprY = nullptr;
+        App::Expression* newExprYaw = nullptr;
+        if (axis == 0) { // normal mostly X
+            // values
+            std::swap(position.x, position.y);
+            position.x = -position.x;
+            rotation.setYawPitchRoll(yaw + 90, pitch, roll);
+
+            // expressions
+            if (exprX) {
+                newExprY = App::ExpressionParser::parse(owner, exprX->toString().c_str());
+            }
+            if (exprY) {
+                std::string expr = "-1 * (" + exprY->toString() + ")";
+                newExprX = App::ExpressionParser::parse(owner, expr.c_str());
+            }
+            if (exprYaw) {
+                std::string expr = "(" + exprYaw->toString() + ") + 90 deg";
+                newExprYaw = App::ExpressionParser::parse(owner, expr.c_str());
+            }
+        }
+        else if (axis == 1) { // normal mostly Y
+            // values
+            std::swap(position.x, position.y);
+            position.y = -position.y;
+            rotation.setYawPitchRoll(yaw - 90, pitch, roll);
+
+            // expressions
+            if (exprX) {
+                std::string expr = "-1 * (" + exprX->toString() + ")";
+                newExprY = App::ExpressionParser::parse(owner, expr.c_str());
+            }
+            if (exprY) {
+                newExprX = App::ExpressionParser::parse(owner, exprY->toString().c_str());
+            }
+            if (exprYaw) {
+                std::string expr = "(" + exprYaw->toString() + ") - 90 deg";
+                newExprYaw = App::ExpressionParser::parse(owner, expr.c_str());
+            }
+        }
+        else {
+            // should not happen
+            return;
+        }
+
+        // store updated placement and expressions back to the document object
+
+        // expressions
+        owner->ExpressionEngine.setValue(oidX, std::move(App::ExpressionPtr(newExprX)));
+        owner->ExpressionEngine.setValue(oidY, std::move(App::ExpressionPtr(newExprY)));
+        owner->ExpressionEngine.setValue(oidYaw, std::move(App::ExpressionPtr(newExprYaw)));
+
+        // values
+        placement.setPosition(position);
+        placement.setRotation(rotation);
+        this->AttachmentOffset.setValue(placement);
+
+    } catch (const Base::Exception& e) {
+        Base::Console().error("Error converting attachment offset of %s: %s\n",
```

Comment:
The error message could be a bit more explanatory:

```suggestion
        Base::Console().error("Error converting legacy attachment offset of %s: %s\n",
```